### PR TITLE
conditionalize shared files

### PR DESCRIPTION
**Pull request by micheleRP**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2560
Review deadline:

## Page previews

- [Intro to rpk](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/get-started/intro-to-rpk/)
- [Install rpk](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/get-started/rpk-install/)
- [rpk Profiles](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/get-started/config-rpk-profile/)
- [Specify Broker Addresses for rpk](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/get-started/broker-admin/)
- [Specify Admin API Addresses for rpk](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/get-started/admin-addresses/)
- [Use Redpanda with the HTTP Proxy API](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/develop/http-proxy/) (core)
- [Use Redpanda with the HTTP Proxy API](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/develop/http-proxy-cloud/) (cloud)
- [Transactions](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/develop/transactions/)
- [Redpanda Schema Registry](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/manage/schema-reg/schema-reg-overview/)
- [Use Schema Registry in Redpanda Console](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/manage/schema-reg/schema-reg-ui/)
- [Use Schema Registry in Redpanda Cloud](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/cloud/schema-reg-cloud/)
- [Use the Schema Registry API](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/manage/schema-reg/schema-reg-api/)
- [Deserialization](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/reference/console/record-deserialization/)
- [Programmable Push Filters](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/reference/console/programmable-push-filters/)
- [Edit Topic Configuration in the Redpanda UI](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/manage/console/edit-topic-configuration/)
- [Deprecated Features](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/upgrade/deprecated/)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Comment by micheleRP** [comment]
@Deflaimun this has the rpk changes we talked about. Would you please review? And @JakeSCahill, would you please check this PR too? This edits all files for Cloud that are single sourced in core, except reference files.

File: modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc
Comment (JakeSCahill) [inline modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc:1]: This is going to be moved over to https://github.com/redpanda-data/cloud-docs?

File: modules/develop/pages/consume-data/follower-fetching.adoc
Comment (JakeSCahill) [inline modules/develop/pages/consume-data/follower-fetching.adoc]: Is the plan to single-source this whole page over to RP Cloud? If so, we should remove the tabs and conditionally render the self-hosted docs and the cloud docs in the right context.

File: modules/develop/pages/http-proxy-cloud.adoc
Comment (JakeSCahill) [inline modules/develop/pages/http-proxy-cloud.adoc:1]: this should live in https://github.com/redpanda-data/cloud-docs

So we don't need the extra `cloud` suffix in the filename.

File: modules/manage/pages/console/edit-topic-configuration.adoc
Comment (JakeSCahill) [inline modules/manage/pages/console/edit-topic-configuration.adoc]: Now that we are going with the strategy of including all relevant content directly in cloud docs, we need to conditionalize this so that self-hosted users see 'Redpanda Console' and cloud users see 'Redpanda UI'.

I've added an attribute that we can use for this to https://github.com/redpanda-data/cloud-docs/pull/3

For example:

```asciidoc
ifdef::env-cloud[]
:ui: Redpanda Cloud
endif::[]
ifndef::env-cloud[]
:ui: Redpanda Console
endif::[]
= Edit Topic Configuration in {ui}
```

File: modules/manage/partials/schema-reg-ui.adoc
Comment (JakeSCahill) [inline modules/manage/partials/schema-reg-ui.adoc:11]: I wouldn't use Console as the environment. Instead we want this to be displayed in all contexts that aren't Redpanda Cloud, so:

```asciidoc
ifndef::env-cloud[]
== Prerequisites

Redpanda Console is automatically configured with Redpanda Cloud deployments. However, if you're running Redpanda Console with self-hosted Redpanda, you must add a valid `schemaRegistry` configuration in Redpanda Console:

.redpanda-console-config.yaml
[,yaml]
----
kafka:
  schemaRegistry:
    enabled: true
    urls: ["https://my-schema-registry.com"]
    username: console
    password: redacted # Or set using flags or env variable
  # To enable protobuf support
  protobuf:
    enabled: true
    schemaRegistry:
      enabled: true
      refreshInterval: 5m
----
endif::[]
```

**Review by JakeSCahill — COMMENTED** [review]
I linked up this PR with the one that includes the content in cloud: https://github.com/redpanda-data/cloud-docs/pull/3

We still have a bunch of broken links.

It seems like the majority of them are coming from links to pages that we don't port over to cloud (because I assume they aren't required). These links are included in the single-sourced content. So we need to do some conditionalization where links get added only when not in the cloud context.

```asciidoc
# Include only in cloud
ifdef::env-cloud[]
endif::[]
# Include only outside of cloud
ifndef::env-cloud[]
endif::[]
```

File: modules/reference/pages/console/record-deserialization.adoc
Comment (JakeSCahill) [inline modules/reference/pages/console/record-deserialization.adoc]: I would undo these changes. This page is about how Redpanda Console deserializes messages.

If we want to include these details in the cloud docs, we should conditionalize the product name and exclude all the self-hosted configuration sections.

File: modules/reference/pages/console/programmable-push-filters.adoc
Comment (JakeSCahill) [inline modules/reference/pages/console/programmable-push-filters.adoc]: This is specifically about Redpanda Console. We should undo these changes. If this page needs to be included in cloud docs, conditionalize the content.

File: modules/manage/partials/schema-reg-ui.adoc
Comment (JakeSCahill) [inline modules/manage/partials/schema-reg-ui.adoc]:
```suggestion
You must add a valid `schemaRegistry` configuration in Redpanda Console:
```

Comment (JakeSCahill) [reply modules/manage/partials/schema-reg-ui.adoc]: This will only get included outside of cloud.

File: modules/manage/partials/schema-reg-ui.adoc
Comment (JakeSCahill) [inline modules/manage/partials/schema-reg-ui.adoc]:
```suggestion
ifndef::env-cloud[]
```

Comment (JakeSCahill) [reply modules/manage/partials/schema-reg-ui.adoc]: I wouldn't use Console as the environment. Instead we want this to be displayed in all contexts that aren't Redpanda Cloud.

File: modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc
Comment (micheleRP) [reply modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc:1]: Yes, it should get moved over there, but it can stay here until we merge cloud-docs, right? Same with the Use HTTP Proxy page, right?

File: modules/develop/pages/consume-data/follower-fetching.adoc
Comment (micheleRP) [reply modules/develop/pages/consume-data/follower-fetching.adoc]: Yes, this page works fine right now for both cloud/core, so I was planning to wait until after we merge cloud docs to conditionalize it with a separate page for cloud.

File: modules/develop/pages/http-proxy-cloud.adoc
Comment (micheleRP) [reply modules/develop/pages/http-proxy-cloud.adoc:1]: Yes, this will live in cloud-docs, but right now it's in docs. I thought we move stuff like this after we merge cloud-docs, no? Can you please remind me of the naming convention we should use? We have http-proxy.adoc + now a partial named http-proxy.adoc and http-proxy-cloud.adoc. What process/naming do you recommend?

File: modules/reference/pages/console/programmable-push-filters.adoc
Comment (micheleRP) [reply modules/reference/pages/console/programmable-push-filters.adoc]: No Vlado confirmed that it's not specific to console and that this could be written more generically for both cloud & console.

File: modules/develop/pages/consume-data/follower-fetching.adoc
Comment (JakeSCahill) [reply modules/develop/pages/consume-data/follower-fetching.adoc]: We should work on this PR and the cloud-docs PR simultaneously. Both are required to achieve single-sourcing.

File: modules/develop/pages/http-proxy-cloud.adoc
Comment (JakeSCahill) [reply modules/develop/pages/http-proxy-cloud.adoc:1]: if the file is being moved to cloud-docs, I'd say we don't need the prefix.

File: modules/reference/pages/console/programmable-push-filters.adoc
Comment (JakeSCahill) [reply modules/reference/pages/console/programmable-push-filters.adoc]: But the plan is to have both self-hosted and cloud docs sourced from the same content. We can use attributes to reference the product name so that it's more helpful for both users. `{ui}` could be replaced by Redpanda Console in SH docs and Redpanda Cloud in cloud docs.

Also, this is now incorrect. It's Redpanda Console's filter engine. You're only filtering messages on Console's side. For cloud docs this could say Redpanda Cloud or we could just remove 'Redpanda' and refer to the filter engine.

> You can use Redpanda filter engine

File: modules/manage/pages/schema-reg/schema-reg-overview.adoc
Comment (JakeSCahill) [inline modules/manage/pages/schema-reg/schema-reg-overview.adoc]: The consensus was to avoid sending users to Redpanda docs from the cloud docs. So, we should single-source this page and avoid specifying the `ROOT` component in links.

File: modules/manage/pages/schema-reg/schema-reg-overview.adoc
Comment (JakeSCahill) [inline modules/manage/pages/schema-reg/schema-reg-overview.adoc]: Same as above

File: modules/manage/pages/schema-reg/schema-reg-ui.adoc
Comment (JakeSCahill) [inline modules/manage/pages/schema-reg/schema-reg-ui.adoc]:
```suggestion
```

File: modules/develop/pages/produce-data/configure-producers.adoc
Comment (JakeSCahill) [inline modules/develop/pages/produce-data/configure-producers.adoc:234]: @micheleRP we probably need a separate issue for this. Do you know if there are restrictions/what they are?

File: modules/manage/pages/schema-reg/schema-reg-api.adoc
Comment (JakeSCahill) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]: We want some conditional content here. For Redpanda Cloud users, we want to link to how to create a cloud cluster.

Comment (JakeSCahill) [reply modules/manage/pages/schema-reg/schema-reg-api.adoc]: No need for `ROOT` because the link will only be used in the SH context.

File: modules/manage/pages/schema-reg/schema-reg-api.adoc
Comment (JakeSCahill) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]: No need for `ROOT` in any of these links other than API docs. They should all be single-sourced into cloud docs.

File: modules/reference/pages/properties/cluster-properties.adoc
Comment (JakeSCahill) [inline modules/reference/pages/properties/cluster-properties.adoc:3]: We need to keep all these aliases.

File: modules/manage/pages/schema-reg/schema-reg-api.adoc
Comment (asimms41) [reply modules/manage/pages/schema-reg/schema-reg-api.adoc]: @JakeSCahill and @micheleRP. I've added a conditional block for Cloud content which directs users to Serverless to create a cluster if they don't have one. Is this right?

File: modules/develop/pages/produce-data/configure-producers.adoc
Comment (asimms41) [reply modules/develop/pages/produce-data/configure-producers.adoc:234]: @JakeSCahill and @micheleRP,  I've added a condition to exclude this subsection from the Cloud docs. Is there a separate pull request/issue for other mentions of the Admin API in single-sourced content?

File: modules/manage/pages/schema-reg/schema-reg-ui.adoc
Comment (asimms41) [reply modules/manage/pages/schema-reg/schema-reg-ui.adoc]: Updated.

File: modules/reference/pages/console/programmable-push-filters.adoc
Comment (asimms41) [reply modules/reference/pages/console/programmable-push-filters.adoc]: @micheleRP and @JakeSCahill, I've added the {ui} attribute in all cases except for the final reference, which now says 'You can use the filter engine.'

File: modules/reference/pages/console/record-deserialization.adoc
Comment (asimms41) [reply modules/reference/pages/console/record-deserialization.adoc]: @JakeSCahill and @micheleRP, I've replaced the product name with the {ui} attribute and added a conditional paragraph in the [Schema Registry section](https://deploy-preview-586--redpanda-docs-preview.netlify.app/current/reference/console/record-deserialization/#schema-registry).

File: modules/manage/pages/console/edit-topic-configuration.adoc
Comment (asimms41) [reply modules/manage/pages/console/edit-topic-configuration.adoc]: @JakeSCahill and @micheleRP. {ui} attribute added.

File: modules/develop/pages/http-proxy-cloud.adoc
Comment (asimms41) [reply modules/develop/pages/http-proxy-cloud.adoc:1]: @micheleRP, There are 2 files with the same content http-proxy.adoc and http-proxy-cloud.adoc. Is that correct?

File: modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc
Comment (asimms41) [reply modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc:1]: @micheleRP, There are 2 files with the same content http-proxy.adoc and http-proxy-cloud.adoc. Is that correct?

File: modules/develop/pages/consume-data/follower-fetching.adoc
Comment (asimms41) [reply modules/develop/pages/consume-data/follower-fetching.adoc]: @JakeSCahill. File has been updated.

File: modules/develop/pages/produce-data/configure-producers.adoc
Comment (JakeSCahill) [inline modules/develop/pages/produce-data/configure-producers.adoc]:
```suggestion
```

File: modules/develop/pages/transactions.adoc
Comment (JakeSCahill) [inline modules/develop/pages/transactions.adoc:16]: cc @micheleRP

File: modules/develop/pages/transactions.adoc
Comment (JakeSCahill) [inline modules/develop/pages/transactions.adoc:40]: We need to exclude or rewrite some of this content for Cloud. We don't want to mention cluster properties or 'self-hosted deployment'.

File: modules/get-started/pages/admin-addresses.adoc
Comment (JakeSCahill) [inline modules/get-started/pages/admin-addresses.adoc:60]: If this page is going into Cloud, we should rewrite/exclude some of this content.  Cloud users don't get access to the `redpanda.yaml`.

@micheleRP would cloud users need to configure `rpk` at all? Or do they just `rpk cloud login`?

File: modules/get-started/pages/broker-admin.adoc
Comment (JakeSCahill) [inline modules/get-started/pages/broker-admin.adoc:7]: Same comment as https://github.com/redpanda-data/docs/pull/586/files#r1677854433

File: modules/manage/pages/schema-reg/schema-reg-api.adoc
Comment (JakeSCahill) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc:940]: We should conditionalize these links. Cloud docs should have links to node property configs or node-configuration-sample.adoc for example.

File: modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc
Comment (JakeSCahill) [inline modules/deploy/pages/deployment-option/cloud/schema-reg-cloud.adoc:3]:
```suggestion
:page-categories: Management, Schema Registry
:ui: Redpanda Cloud
```

File: modules/deploy/pages/deployment-option/cloud/create-topic.adoc
Comment (JakeSCahill) [inline modules/deploy/pages/deployment-option/cloud/create-topic.adoc:4]: Why did we remove the alias? `:page-aliases: cloud:create-topic.adoc`

File: modules/develop/pages/produce-data/configure-producers.adoc
Comment (JakeSCahill) [reply modules/develop/pages/produce-data/configure-producers.adoc:234]: We should handle all conditionalization in this PR.

File: modules/develop/pages/transactions.adoc
Comment (JakeSCahill) [reply modules/develop/pages/transactions.adoc:16]: Did we find out if remote recovery is enabled @micheleRP ?

File: modules/reference/pages/rpk/rpk-debug/rpk-debug.adoc
Comment (JakeSCahill) [inline modules/reference/pages/rpk/rpk-debug/rpk-debug.adoc:3]: We shouldn't be removing any page aliases.

File: modules/reference/pages/rpk/rpk-generate/rpk-generate.adoc
Comment (JakeSCahill) [inline modules/reference/pages/rpk/rpk-generate/rpk-generate.adoc:3]: Please put the page alias back.

File: modules/reference/pages/rpk/rpk-group/rpk-group.adoc
Comment (JakeSCahill) [inline modules/reference/pages/rpk/rpk-group/rpk-group.adoc:3]: The page alias still needs to exist. Put it above the tag.

File: modules/reference/pages/rpk/rpk-plugin/rpk-plugin.adoc
Comment (JakeSCahill) [inline modules/reference/pages/rpk/rpk-plugin/rpk-plugin.adoc:4]: The page alias still needs to exist. Put it above the tag.

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-info.adoc
Comment (JakeSCahill) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-info.adoc:3]: We need all existing page aliases. Please add this back above the tag.

File: modules/deploy/pages/deployment-option/cloud/create-topic.adoc
Comment (micheleRP) [reply modules/deploy/pages/deployment-option/cloud/create-topic.adoc:4]: idk, I've forgotten why I did or didn't do things along the way. But it's back now!

**Review by JakeSCahill — APPROVED** [review]
LGTM

@Deflaimun we should do a rebase for the beta branch after we merge this.

**Pull request merged**
